pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -24,6 +24,40 @@ const (
 	PausedAnnotation = "baremetalhost.metal3.io/paused"
 )
 
+// RootDeviceHints holds the hints for specifying the storage location
+// for the root filesystem for the image.
+type RootDeviceHints struct {
```

Comment:
This is mostly identical to the existing `Storage` struct, and many of the differences are not improvements. It would be worth investigating the feasibility of just reusing that struct.

Reply:
Related to above size issue and proposal to use subset in metal3-doc discussion I would like to keep separate structs.  OK?